plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/AllFilesTable.java

```diff
@@ -0,0 +1,172 @@
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class AllFilesTable implements SystemTable {
+    private final ConnectorTableMetadata tableMetadata;
+    private final Table icebergTable;
+
+    public AllFilesTable(SchemaTableName tableName, TypeManager typeManager, Table icebergTable)
+    {
+        this.icebergTable = requireNonNull(icebergTable, "icebergTable is null");
+        this.tableMetadata = new ConnectorTableMetadata(requireNonNull(tableName, "tableName is null"),
+                ImmutableList.<ColumnMetadata>builder()
+                        .add(new ColumnMetadata("content", INTEGER))
+                        .add(new ColumnMetadata("file_path", VARCHAR))
+                        .add(new ColumnMetadata("file_format", VARCHAR))
+                        .add(new ColumnMetadata("record_count", BIGINT))
+                        .add(new ColumnMetadata("file_size_in_bytes", BIGINT))
+                        .add(new ColumnMetadata("column_sizes", typeManager.getType(mapType(INTEGER.getTypeSignature(), BIGINT.getTypeSignature()))))
+                        .add(new ColumnMetadata("value_counts", typeManager.getType(mapType(INTEGER.getTypeSignature(), BIGINT.getTypeSignature()))))
+                        .add(new ColumnMetadata("null_value_counts", typeManager.getType(mapType(INTEGER.getTypeSignature(), BIGINT.getTypeSignature()))))
+                        .add(new ColumnMetadata("nan_value_counts", typeManager.getType(mapType(INTEGER.getTypeSignature(), BIGINT.getTypeSignature()))))
+                        .add(new ColumnMetadata("lower_bounds", typeManager.getType(mapType(INTEGER.getTypeSignature(), VARCHAR.getTypeSignature()))))
+                        .add(new ColumnMetadata("upper_bounds", typeManager.getType(mapType(INTEGER.getTypeSignature(), VARCHAR.getTypeSignature()))))
+                        .add(new ColumnMetadata("key_metadata", VARBINARY))
+                        .add(new ColumnMetadata("split_offsets", new ArrayType(BIGINT)))
+                        .add(new ColumnMetadata("equality_ids", new ArrayType(INTEGER)))
+                        .build());
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        return new FixedPageSource(buildPages());
+    }
+
+    @Override
+    public Distribution getDistribution() {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata() {
+        return tableMetadata;
+    }
+
+    protected List<Page> buildPages(long snapshotId)
+    {
+        PageListBuilder pagesBuilder = PageListBuilder.forTable(tableMetadata);
+        Map<Integer, Type> idToTypeMapping = getIcebergIdToTypeMapping(icebergTable.schema());
+
+        TableScan tableScan = icebergTable.newScan()
```

Comment:
What if instead of implementing the logic of retrieving the data files ourselves, we rely on iceberg for this?

```
Table allDataFilesTable = MetadataTableUtils.createMetadataTableInstance(icebergTable, MetadataTableType.ALL_DATA_FILES);
```

The result should be the same right?

Reply:
yes, @findinpath result should be the same.

I also thought that eventually, we would be able to use the Iceberg's API for all the metadata tables, but was not sure if we don't want to depend on all the internal utils. Also, it won't to very straight fwd to build the schema, and the result using the Iceberg metadata tables. Another point, I thought was not sure if there is some historical reason to implement in Trino this way, so followed this way.

Actually, I had mentioned this point in the other issue as well, I had created for `$all_entries` table: https://github.com/trinodb/trino/issues/10882

from the preceding issue:

Also wondering if we can also think of reusing Metadata table classes that Iceberg has like [AllEntries](https://github.com/apache/iceberg/blob/5505321af663befa89095ed6643b2de9be481788/core/src/main/java/org/apache/iceberg/AllEntriesTable.java#L110) instead of doing it doing it in the Trino APIs.

So, maybe in the coming days, with more knowledge about the current state, history, we will be able to upgrade Trino's metadata tables to use the Iceberg metadata directly.

thanks.